.env

```diff
@@ -0,0 +1,6 @@
+USERNAME=
```

Comment:
I think that we'd want to keep this in some file with a `.template` post-fix or similar. If we keep `.env` committed to source control, it makes it likely that someone (probably myself 🤣) will accidentally push their changes to a branch and possibly leak a secret.

Whatever the non-template file name, we should ensure it's in the `.gitignore` so that it doesn't get pushed up.

Reply:
Ah, yeah. I wondered about that but forgot to look that up.

Any idea what best practice is for .env files? Some thoughts:
1. Update readme to tell users to create an environment file and paste their creds into it (and .gitignore all .env*)
2. Update run.* scripts to look for .env file and create/populate it with prompts

Also about .template - sounds like all .env files should be gitignored one way or another?

Reply:
My usual go to is this utility, which has decently wide adoption: https://direnv.net/. Does not work on Windows, though :/

Typically I've seen the following practices:

`.env.template` - guidance for user to copy to file called `.env` or similar, filled out, and then it can be export by sourcing that file. On `*nix` this looks like:

```bash
cp .env.template .env
vim .env // fill out falues
. .env
```

And the `.gitignore` would include:

```bash
.env
.envrc
```

but *not* `.env.template` - this file is a source for people to use and may change over time. It won't have any credentials, so it's safe to commit.

Today, the `.gitignore` only has `.envrc` because that's what I use personally (see direnv tool) but no reason it can't also include `.env`

Reply:
Okay, will probably be a separate issue to do this in a cross platform way that doesn't leak creds.

Until then, I've re-added the .env.template file and updated the readme to say "fill out the env.template".
Maybe that's not the best approach for the readme, up to you.

Reply:
I've also briefly read something about multi stage builds on docker as a way to get protected files into your build without them showing up in logs (though the container is deleted now).

Maybe run script creates .env file if not present, echo .env.template into it, and opens vim.
Then docker can copy that .env file into the build process, source the env file, and the run script can delete the local copy of .env

Reply:
The `dockerignore` file - https://github.com/nmiodice/id4-reservation-tracker/blob/main/.dockerignore - should prevent those environment files from being added to the built image. docs: https://docs.docker.com/engine/reference/builder/#dockerignore-file

And multistage builds will help reduce the final image size. Since we don't need to push it to a docker registry, it won't help us here. Whatever files were copied into the intermediate build images will still live in some cached docker layer on your system whether or not you use a multistage build.

Reply:
>Maybe run script creates .env file if not present, echo .env.template into it, and opens vim.
Then docker can copy that .env file into the build process, source the env file, and the run script can delete the local copy of .env

I'd shy away from this - with Docker, typically environment variables are the standard way to provide these types of configuration. More generally (not really an issue for _this_ project, but Docker/containers in general) leveraging environment variables makes it easier to run containers in a variety of different platform.

For example, kubernetes or Azure Container Instances will all use environment variables, not `.env` files, to configure containers.

So `.env` style approaches are common with docker and I'd prefer to not automate much of it in this project. It's tough to do cross platform, and everyone does it differently :/